src/ui/frmpreferences.cpp

```diff
@@ -92,10 +92,15 @@ void frmPreferences::on_treeWidget_currentItemChanged(QTreeWidgetItem *current,
 
 void frmPreferences::on_buttonBox_accepted()
 {
-    if(m_keyGrabber->findConflicts()){
+    if(m_keyGrabber->hasConflicts()){
+        //Try our best to show the error to the user immediately.
+        ui->treeWidget->setCurrentItem( ui->treeWidget->topLevelItem(4) );
```

Comment:
What's the UI reference? I can't use ui->page_4 because this is setting the item in the left-hand list, rather than changing the page of the stackedWidget. I don't think they have any references.

I do agree that using magic numbers here sucks. But getting rid of them is a bit more work unless I'm missing something.